packages/babel-helpers/src/helpers/applyDecs.js

```diff
@@ -0,0 +1,728 @@
+/* @minVersion 7.16.6 */
+
+/**
+  Enums are used in this file, but not assigned to vars to avoid non-hoistable values
+
+  CONSTRUCTOR = 0;
+  PUBLIC = 1;
+  PRIVATE = 2;
+
+  FIELD = 0;
+  ACCESSOR = 1;
+  METHOD = 2;
+  GETTER = 3;
+  SETTER = 4;
+
+  STATIC = 5;
+*/
+
+function createMetadataMethodsForProperty(metadataMap, kind, property) {
+  return {
+    getMetadata(key) {
+      if (typeof key !== "symbol") {
+        throw new TypeError("Metadata keys must be symbols, received: " + key);
+      }
+
+      var metadataForKey = metadataMap[key];
+
+      if (metadataForKey === void 0) return void 0;
+
+      if (kind === 1 /* PUBLIC */) {
+        var pub = metadataForKey.public;
+        if (pub !== void 0) {
+          return pub[property];
+        }
+      } else if (kind === 2 /* PRIVATE */) {
+        var priv = metadataForKey.private;
+        if (priv !== void 0) {
+          return priv.get(property);
+        }
+      } else if (Object.hasOwnProperty.call(metadataForKey, "constructor")) {
+        return metadataForKey.constructor;
+      }
+    },
+    setMetadata(key, value) {
+      if (typeof key !== "symbol") {
+        throw new TypeError("Metadata keys must be symbols, received: " + key);
+      }
+
+      var metadataForKey = metadataMap[key];
+
+      if (metadataForKey === void 0) {
+        metadataForKey = metadataMap[key] = {};
+      }
+
+      if (kind === 1 /* PUBLIC */) {
+        var pub = metadataForKey.public;
+
+        if (pub === void 0) {
+          pub = metadataForKey.public = Object.create(null);
+        }
+
+        pub[property] = value;
+      } else if (kind === 2 /* PRIVATE */) {
+        var priv = metadataForKey.priv;
+
+        if (priv === void 0) {
+          priv = metadataForKey.private = new Map();
+        }
+
+        priv.set(property, value);
+      } else {
+        metadataForKey.constructor = value;
+      }
+    },
+  };
+}
+
+function convertMetadataMapToFinal(obj, metadataMap) {
+  var parentMetadataMap = obj[Symbol.metadata];
```

Comment:
If this is the only thing that needs to be polyfilled, we could use `Symbol.metadata || Symbol.for("Symbol.metadata")` everywhere.

Reply:
Yes, this is the only thing, so we could do that